gson/src/test/java/com/google/gson/GsonTest.java

```diff
@@ -82,4 +86,16 @@ public void testClonedTypeAdapterFactoryListsAreIndependent() {
     }
     @Override public Object read(JsonReader in) throws IOException { return null; }
   }
+
+  public void testNullAdapterLookup() {
+    // Formalize the handling of null type tokens (previously it was partially handled, but still throwing an error)
```

Comment:
Not sure if this comment line here is useful in the test code. To me this message only seems to be relevant regarding why this pull request or commit was created, but not why this test exists.

---

Note that unlike eamonnmcmanus, I am not a member of this project. Feel free to consider my comments only as suggestion.